site-manage/site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/rsf/GroupListProducer.java

```diff
@@ -126,17 +133,29 @@ public void fillComponents(UIContainer tofill, ViewParameters viewparams,
 				String groupId = group.getId();
 				UIBranchContainer grouprow = UIBranchContainer.make(deleteForm, "group-row:", group.getId());
 				
-				UIOutput.make(grouprow, "group-title-label", group.getTitle());
+				String groupTitle = Validator.escapeHtml(group.getTitle());
+				if (group.isLocked()) {
+					UIOutput groupIcon = UIOutput.make(grouprow, "group-icon");
+					groupIcon.decorate(new UIStyleDecorator("fa-lock"));
+					groupIcon.decorate(new UITooltipDecorator(messageLocator.getMessage("group.locked")));
+				}
+                                
+				UIOutput.make(grouprow, "group-title-label", groupTitle);
 				UIInput name =
-						UIInput.make(grouprow, "group-name-input", "#{SitegroupEditHandler.nil}", group.getTitle());
+						UIInput.make(grouprow, "group-name-input", "#{SitegroupEditHandler.nil}", groupTitle);
 				UIOutput nameLabel =
 						UIOutput.make(grouprow, "group-name-label", messageLocator.getMessage("group.title"));
 				
 				nameLabel.decorate(new UILabelTargetDecorator(name));
-				UIInternalLink editLink = UIInternalLink.make(grouprow,"group-title",group.getTitle(),
+				UIInternalLink editLink = UIInternalLink.make(grouprow,"group-title", groupTitle,
 																					 new GroupEditViewParameters(GroupEditProducer.VIEW_ID, groupId));
-				editLink.decorators = new DecoratorList(new UITooltipDecorator(messageLocator.getMessage("editgroup.revise")+ ":" + group.getTitle()));
-				
+				if (group.isLocked()) {
+					editLink.decorators = new DecoratorList(new UITooltipDecorator(messageLocator.getMessage("editgroup.noteditable")+ ":" + groupTitle));
+				} else {
+					editLink.decorators = new DecoratorList(new UITooltipDecorator(messageLocator.getMessage("editgroup.revise")+ ":" + groupTitle));
+				}
+				M_log.debug("Check if the group is locked : "+group.getId()+" -> "+group.isLocked());
```

Comment:
M_log.debug("Check if the group is locked : {} -> {}", group.getId(), group.isLocked());